### ISSUE TITLE
Imported state for aws_elasticache_cluster causes a problem with the port field

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#341]
_This issue was originally opened by @makered as hashicorp/terraform#9796. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi,

After importing my Redis EC cluster via:

`terraform import aws_elasticache_cluster.my_cluster my_cluster`

and writing a resource script to match the state that contains:

`port = 6379`

the plan for some reason shows:

`port: "" => "6379" (forces new resource)`

The state file contains only the following in relation to the port:

`"cache_nodes.0.port": "6379"`

When I manually added the following to the state file:

`"port": "6379"`

and re-ran the plan, the result does not show a change on port and there is no plan to replace the cluster

Could this be a bug?

I am running Terraform 0.7.7

**Comment by pmacdougall** [comment]
Ran into this with `Terraform v0.9.8` still.  
Will try manually editing the state file to see if I can make progress with my project.

**Comment by pmacdougall** [comment]
Actually, my issue was that I was using `aws_elasticache_cluster`, when I should have been using `aws_elasticache_replication_group`.   When there's only 1 node these look and behave quite similarly it seems.

**Comment by leszekeljasz** [comment]
This looks like a bug, and it still reproduces with the most recent versions of terraform and aws provider.
```
$ terraform version
Terraform v0.11.2
+ provider.aws v1.7.1
```

Steps to replicate:

### main.tf
```
provider "aws" {
    region = "eu-central-1"
}

data "aws_availability_zones" "available" {}

resource "aws_vpc" "myvpc" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_subnet" "sub_a" {
    vpc_id	= "${aws_vpc.myvpc.id}"
    cidr_block	= "10.0.0.0/24"
    availability_zone = "${data.aws_availability_zones.available.names[0]}"
}

resource "aws_subnet" "sub_b" {
    vpc_id      = "${aws_vpc.myvpc.id}"
    cidr_block  = "10.0.1.0/24"
    availability_zone = "${data.aws_availability_zones.available.names[1]}"
}

resource "aws_security_group" "sec_gr" {
    name_prefix = "test_"
    vpc_id      = "${aws_vpc.myvpc.id}"
}

resource "aws_elasticache_subnet_group" "sub_gr" {
    name = "test-20180123152301"
    subnet_ids  = [
        "${aws_subnet.sub_a.id}",
        "${aws_subnet.sub_b.id}"
    ]
}

resource "aws_elasticache_cluster" "redis" {
    cluster_id              = "test-20180123152301"
    engine                  = "redis"
    engine_version          = "3.2"
    node_type               = "cache.t2.micro"
    num_cache_nodes         = 1
    parameter_group_name    = "default.redis3.2"
    port                    = 6379
    subnet_group_name       = "${aws_elasticache_subnet_group.sub_gr.name}"
    security_group_ids      = [ "${aws_security_group.sec_gr.id}" ]
    depends_on              = [ "aws_elasticache_subnet_group.sub_gr" ]
}
```

1. `terraform apply`
2. `terraform state rm aws_elasticache_cluster.redis`
3. `terraform import aws_elasticache_cluster.redis test-20180123152301`
4. `terraform plan`   # asks to recreate redis cluster:
```
$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

aws_vpc.myvpc: Refreshing state... (ID: vpc-89cc1ce2)
data.aws_availability_zones.available: Refreshing state...
aws_subnet.sub_a: Refreshing state... (ID: subnet-6b88ed00)
aws_security_group.sec_gr: Refreshing state... (ID: sg-886d6ee2)
aws_subnet.sub_b: Refreshing state... (ID: subnet-7cc71b01)
aws_elasticache_subnet_group.sub_gr: Refreshing state... (ID: test-20180123152301)
aws_elasticache_cluster.redis: Refreshing state... (ID: test-20180123152301)

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ aws_elasticache_cluster.redis (new resource required)
      id:                            "test-20180123152301" => <computed> (forces new resource)
      apply_immediately:             "" => <computed>
      availability_zone:             "eu-central-1a" => <computed>
      az_mode:                       "single-az" => <computed>
      cache_nodes.#:                 "1" => <computed>
      cluster_address:               "" => <computed>
      cluster_id:                    "test-20180123152301" => "test-20180123152301"
      configuration_endpoint:        "" => <computed>
      engine:                        "redis" => "redis"
      engine_version:                "3.2.10" => "3.2"
      maintenance_window:            "wed:02:30-wed:03:30" => <computed>
      node_type:                     "cache.t2.micro" => "cache.t2.micro"
      num_cache_nodes:               "1" => "1"
      parameter_group_name:          "default.redis3.2" => "default.redis3.2"
      port:                          "" => "6379" (forces new resource)
      replication_group_id:          "" => <computed>
      security_group_ids.#:          "1" => "1"
      security_group_ids.1962281463: "sg-886d6ee2" => "sg-886d6ee2"
      security_group_names.#:        "0" => <computed>
      snapshot_window:               "" => <computed>
      subnet_group_name:             "test-20180123152301" => "test-20180123152301"


Plan: 1 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```

**Comment by jsnod** [comment]
This is still an issue with Terraform v0.11.3 / provider.aws v1.11.0.  The workaround posted by @hamstah here works: https://github.com/hashicorp/terraform/issues/9796#issuecomment-294907612

**Comment by bflad** [comment]
The fix for this should be merged into master and will be released with v1.13.0 of the AWS provider, likely at the end of next week.

**Comment by bflad** [comment]
This has been released in [version 1.13.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1130-march-28-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
